tee-worker/scripts/litentry/release/prod_deploy.sh

```diff
@@ -0,0 +1,664 @@
+    subcommand_flags=$(cat "$WORKER_CONFIG" | jq -r ".workers[$i].subcommand_flags[]")
+    args=
+    for flag in $flags; do
+      args+=" $flag"
+    done
+    args+=" run"
+    for subcommand_flag in $subcommand_flags; do
+      args+=" $subcommand_flag"
+    done
+    sed -i "s;ARGS;$args;" worker$i.service
+    sudo mv worker$i.service -f /etc/systemd/system/
+  done
+  rm worker.service
+
+  if [ "$WITH_PARACHAIN" = true ]; then
+    sudo cp *.service -f /etc/systemd/system/
+  fi
+
+  cd ~
+  rm -rf "$temp_dir"
+  sudo systemctl daemon-reload
+  echo "Done, please check files under /etc/systemd/system/"
+  echo "Restart the services to take effect"
+}
+
+function stop_services {
+  stop_worker_services
+
+  if [ "$WITH_PARACHAIN" = true ]; then
+    stop_parachain_services
+  fi
+}
+
+function stop_worker_services {
+  echo "Stopping worker services ..."
+  for ((i = 0; i < $WORKER_COUNT; i++)); do
+    sudo systemctl stop "worker$i.service"
+    sleep 5
+  done
+  backup_workers
+}
+
+function backup_workers {
+  echo "Backing up workers ..."
+  now=$(date +"%Y%m%d-%H%M%S")
+  cd "$WORKER_BASEDIR" || exit
+  for i in $(ls -d * 2>/dev/null); do
+    outdir="$WORKER_BACKUP_BASEDIR/$i-$now"
+    cp -rf "$i" "$outdir"
+    echo "Worker backed up into $outdir"
+  done
+}
+
+function stop_parachain_services {
+  echo "Stopping parachain services ..."
+  sudo systemctl stop para-alice.service relay-alice.service relay-bob.service
+  backup_parachain_logs
+}
+
+function backup_parachain_logs {
+  if [[ -f $PARACHAIN_BASEDIR/*.log ]]; then
+    echo "Backing up parachain logs ..."
+    now=$(date +"%Y%m%d-%H%M%S")
+    outdir="$PARACHAIN_LOG_BACKUP_BASEDIR/log-$now"
+    mkdir -p "$outdir"
+    cp "$PARACHAIN_BASEDIR"/*.log "$outdir" || true
+    echo "Parachain logs backed up into $outdir"
+  fi
+}
+
+function prune {
+  if [ "$DISCARD" = true ]; then
+    echo "Pruning the existing state ..."
+    if [ "$WITH_PARACHAIN" = true ]; then
+      echo "remove everything under $PARACHAIN_BASEDIR"
+      rm -rf "$PARACHAIN_BASEDIR"/*
+    fi
+
+    echo "remove everything under $WORKER_BASEDIR"
+    rm -rf "$WORKER_BASEDIR"/*
+  fi
+}
+
+function start_services {
+  if [ "$WITH_PARACHAIN" = true ]; then
+    start_parachain_services
+  fi
+
+  latest_parentchain_block
+  set_scheduled_enclave
+  setup_working_dir
+  update_parachain_start_block_in_service
+  start_worker_services
+  echo "Done"
+}
+
+function start_parachain_services {
+  echo "Restarting parachain services ..."
+
+  cp "$RELEASE_PKGDIR/polkadot" "$PARACHAIN_BASEDIR"
+  chmod a+x "$PARACHAIN_BASEDIR/polkadot"
+  cp "$RELEASE_PKGDIR/litentry-collator" "$PARACHAIN_BASEDIR"
+  chmod a+x "$PARACHAIN_BASEDIR/litentry-collator"
+
+  cd "$PARACHAIN_BASEDIR" || exit
+  ./polkadot build-spec --chain rococo-local --disable-default-bootnode --raw > rococo-local-chain-spec.json
+  ./litentry-collator export-genesis-state --chain $CHAIN-dev > genesis-state
```

Comment:
I had this question earlier, if this is prod, should this `$CHAIN-dev` ?

Reply:
Thanks. I am refactor it, and try to break into pieces, and maybe move to another repo. It's better not give out our release process public.